### PR TITLE
+htc #19708 provides default marshaller for akka.Done (200, empty text…

### LINKED ISSUE #19708: Provide marshaller for akka.Done
for HTTP, should be a empty response with 200.

### PR DESCRIPTION
**Pull request by ktoso**
…t response)

Resolves #19708

### CODE REVIEW

File: akka-http/src/main/scala/akka/http/scaladsl/marshalling/PredefinedToResponseMarshallers.scala [hunk 1]
```diff
@@ -18,7 +20,7 @@ trait PredefinedToResponseMarshallers extends LowPriorityToResponseMarshallerImp
                                   implicit m: ToEntityMarshaller[T]): ToResponseMarshaller[T] =
     fromStatusCodeAndHeadersAndValue compose (t ⇒ (status, headers, t))
 
-  implicit val fromResponse: TRM[HttpResponse] = Marshaller.opaque(conforms)
+  implicit val fromResponse: TRM[HttpResponse] = Marshaller.opaque(ConstantFun.scalaIdentityFunction)
```
Comment (ktoso) [inline akka-http/src/main/scala/akka/http/scaladsl/marshalling/PredefinedToResponseMarshallers.scala]: less sneaky, same thing

File: akka-http/src/main/scala/akka/http/scaladsl/marshalling/PredefinedToEntityMarshallers.scala [hunk 2]
```diff
@@ -35,6 +36,11 @@ trait PredefinedToEntityMarshallers extends MultipartMarshallers {
       HttpEntity(contentType, array)
     } else HttpEntity.Empty
 
+  implicit val DoneMarshaller: ToEntityMarshaller[akka.Done] =
+    Marshaller.withFixedContentType(`text/plain(UTF-8)`) { done ⇒
+      HttpEntity(`text/plain(UTF-8)`, "")
```
Comment (ktoso) [inline akka-http/src/main/scala/akka/http/scaladsl/marshalling/PredefinedToEntityMarshallers.scala]: i.e. `complete(akka.Done)` or `Future[akka.Done]` etc results in

```
HTTP/1.1 200 OK
Content-Length: 0
Content-Type: text/plain; charset=UTF-8
Date: Mon, 08 Feb 2016 15:06:00 GMT
Server: akka-http/2.4-SNAPSHOT

```

File: akka-http/src/main/scala/akka/http/scaladsl/marshalling/PredefinedToResponseMarshallers.scala (hunk 1 above)
Comment (viktorklang) [inline akka-http/src/main/scala/akka/http/scaladsl/marshalling/PredefinedToResponseMarshallers.scala]: ALL THE FUN ALL THE TIME

**Comment by johanandren** [comment]
LGTM

**Comment by johanandren** [comment]
PLS BUILD

**Comment by ktoso** [comment]
Failure unrelated, logged it here https://github.com/akka/akka/issues/19725

**Comment by ktoso** [comment]
PLS BUILD

**Pull request merged**